### PR TITLE
Remove JsonViewComponentResult

### PR DESCRIPTION
**Pull request by ryanbrandenburg**
Remove JsonViewComponentResult and its tests.

### CODE REVIEW

File: src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewComponent.cs [hunk 1]
```diff
@@ -13,6 +13,7 @@
 using Microsoft.AspNetCore.Mvc.ViewFeatures.Internal;
 using Microsoft.AspNetCore.Routing;
 using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
```
Comment (dougbu) [inline src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewComponent.cs]: Why?

File: test/WebSites/BasicWebSite/Controllers/HomeController.cs [hunk 2]
```diff
@@ -83,11 +83,6 @@ public IActionResult JsonHelperWithSettingsInView()
             return View(person);
         }
 
-        public IActionResult JsonTextInView()
-        {
-            return View();
```
Comment (dougbu) [inline test/WebSites/BasicWebSite/Controllers/HomeController.cs]: Also remove the `JsonTextInView.cshtml` file.

File: test/WebSites/TagHelpersWebSite/TagHelpers/TagCloudViewComponentTagHelper.cs [hunk 3]
```diff
@@ -55,15 +61,18 @@ public void Init(TagHelperContext context)
             {
                 TypeInfo = typeof(TagCloudViewComponentTagHelper).GetTypeInfo(),
                 ShortName = "TagCloudViewComponentTagHelper",
-                FullName = "TagCloudViewComponentTagHelper",
+                FullName = "TagCloudViewComponentTagHelper"
```
Comment (dougbu) [inline test/WebSites/TagHelpersWebSite/TagHelpers/TagCloudViewComponentTagHelper.cs]: Revert, we like these trailing commas

File: test/WebSites/TagHelpersWebSite/TagHelpers/TagCloudViewComponentTagHelper.cs [hunk 4]
```diff
@@ -72,7 +81,18 @@ public void Init(TagHelperContext context)
...
+
+            var context = ViewContext ?? ViewComponentContext.ViewContext;
+
+            var serializerSettings =
+                    context
+                    .HttpContext
+                    .RequestServices
+                    .GetRequiredService<IOptions<MvcJsonOptions>>()
+                    .Value
+                    .SerializerSettings;
```
Comment (dougbu) [inline test/WebSites/TagHelpersWebSite/TagHelpers/TagCloudViewComponentTagHelper.cs]: Fix odd indentation.

File: test/WebSites/TagHelpersWebSite/TagHelpers/TagCloudViewComponentTagHelper.cs [hunk 5]
```diff
@@ -72,7 +81,18 @@ public void Init(TagHelperContext context)
         public async Task<IViewComponentResult> InvokeAsync(int count)
         {
             var tags = await GetTagsAsync(count);
-            return new JsonViewComponentResult(tags);
+
+            var context = ViewContext ?? ViewComponentContext.ViewContext;
```
Comment (dougbu) [inline test/WebSites/TagHelpersWebSite/TagHelpers/TagCloudViewComponentTagHelper.cs]: Should be no scenario where `ViewContext` is `null`.

File: test/WebSites/TagHelpersWebSite/TagHelpers/TagCloudViewComponentTagHelper.cs [hunk 6]
```diff
@@ -55,15 +61,18 @@ public void Init(TagHelperContext context)
...
+                FullName = "TagCloudViewComponentTagHelper"
             };
 
-            await result.ExecuteAsync(new ViewComponentContext(
-                viewComponentDescriptor,
-                new Dictionary<string, object>(),
-                _htmlEncoder,
-                ViewContext,
-                writer));
+            var viewComponentContext = ViewComponentContext ??
```
Comment (dougbu) [inline test/WebSites/TagHelpersWebSite/TagHelpers/TagCloudViewComponentTagHelper.cs]: Should be no scenario where `ViewComponentContext` is `null`.

**Comment by dougbu** [comment]
:watch:

**Comment by ryanbrandenburg** [comment]
:up::date:

File: test/WebSites/BasicWebSite/Controllers/HomeController.cs (hunk 2 above)
Comment (dougbu) [inline test/WebSites/BasicWebSite/Controllers/HomeController.cs]: thx

File: test/Microsoft.AspNetCore.Mvc.FunctionalTests/compiler/resources/TagHelpersWebSite.Home.Index.html [hunk 7]
```diff
@@ -33,11 +33,11 @@ <h1 style="font-family: cursive;">ASP.NET vNext - Home Page</h1>
 
 <div>
     <h3 style="font-family: cursive;">Current Tag Cloud from Tag Helper</h3>
-    <div>["Lorem","ipsum","dolor","sit","amet","consectetur","adipisicing","elit","sed","do","eiusmod","tempor","incididunt","ut","labore","et","dolore","magna","aliquaUt","enim"]</div>
+    <div>Lorem,ipsum,dolor,sit,amet,consectetur,adipisicing,elit,sed,do,eiusmod,tempor,incididunt,ut,labore,et,dolore,magna,aliquaUt,enim</div>
```
Comment (dougbu) [inline test/Microsoft.AspNetCore.Mvc.FunctionalTests/compiler/resources/TagHelpersWebSite.Home.Index.html]: :clap:

**Comment by dougbu** [comment]
:shipit:

**Pull request merged**